### DISCUSSION TITLE
Preserving the original swipe control behavior

### DISCUSSION THREAD

**Discussion by Navixer** [ReVanced/revanced-discussions#475] [Question] [answered]
In the original Vanced the right side of the screen is volume control and the left is brightness in full screen mode. It cuts right in the middle and covers the whole screen. The thing is these controls are disabled when you tap the screen to reveal the title and the video controls. And then you can return to portrait mode using swipe down the gesture. The problem with Revanced is it doesn't disable when you tap the screen you can still use the brightness and volume swipes. The area to pull down to minimize the video is a thin column in the middle of the screen. This causes discomfort and irregularity problems, the areas to swipe are not stable either.

 It would be very useful to have the original behavior, 
1) Disabling swipes when tapped, (when video controls are active) 
2) Swipe down to return to portrait mode covers the whole screen. (when swipes are disabled) 

I have searched for similar discussions but couldn't find any. 
Thanks for the attention.

**Comment by clemensschlipfinger** [comment]
disabling swipes when tapped, would be really helpful

**Comment by oSumAtrIX** [comment]
> Disabling swipes when tapped, (when video controls are active)

@shadow578 We are already hooking in YouTube to detect if the controls are being shown or hidden in patches such as in the SponsorBlock patch. Might be useful to consider this.

**Reply by shadow578** [reply]
I'll look into it soon

**Comment by shadow578** [comment]
@Navixer just to confirm, in the second point
> 2. Swipe down to return to portrait mode covers the whole screen. (when swipes are disabled)

you're reffering to behaviour that applies when player controls are visible, right?

Also, this enhancement only applies to non- Press-to-Swipe behaviour (so Vanced- like), correct?

**Reply by Navixer** [reply]
Yes when the player controls are visible. 
Yes it's not about press to swipe. It's only about normal swipe.
Here's the original behavior I'm talking about:
https://streamable.com/6gxkct

**Comment by Navixer** [comment]
This is the behavior of Revanced
https://streamable.com/g7pnqy
Pull down swipe doesn't work*, it gets confused between pull down and volume / brightness swipe. This causes jitter as you see in this video. 

*It actually works but only in the middle.  A small portion like %20 of the screen. It's between those prev and next buttons. I want that area to cover the whole screen.

**Comment by Navixer** [comment, marked as answer]
It looks like it's fixed in the latest patch: YouTube ver 17.36.37. 
All my suggestions work as I described now. Thanks for the good work.

**Reply by Lukerator** [reply]
It does not work for me can you help pls?
